src/shims/mod.rs

```diff
@@ -44,11 +40,13 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
 
         // Try to see if we can do something about foreign items.
         if this.tcx.is_foreign_item(instance.def_id()) {
-            // An external function that we cannot find MIR for, but we can still run enough
+            // An external function that we (possibly) cannot find MIR for, but we can still run enough
```

Comment:
This "(possibly)" is confusing. I assume you added this because we sometimes return a `Body`? Still, the foreign item here is an FFI import, they never have a MIR.

Reply:
No - we load MIR for `__rust_start_panic` and `panic_impl`, hence the `(possibly)`.

Reply:
The external item doesn't have a body itself though, we use another body from elsewhere (basically, "linking").
So what about
```
// An external function call that does not have a MIR body.  We either find MIR elsewhere
// or emulate their effect.
```